department-of-veterans-affairs/caseflow
# Stephan/appeals 7472 v5

Resolves # APPEALS-7472

### Description
This PR resolves the Review and confirm new appeal stream for when a COB or SSC user finds the necessary front end changes.   . This PR incorperates reasons for the split on the front-end component hook, along with items necessary for the new appeal stream. A cancel, back, and split appeal button has been incorporated to the GUI. 

### Acceptance Criteria
- [X] Code compiles correctly SFW 09/21/20022

### Testing Plan
1. Must run bundle install, a migration from master (bin/rails db:migrate RAILS_ENV=development), and make reset. "Seed data is applied so all containers must be up to date with master & feature and make reset to reseed the data."
2. Ensure any intake appeal has 2 or more request issues to split; even if hearing on a 10182. (Maybe DeborahBVAIntake) to intake appeals as necessary.
3. Sign in as one of the Supervisory Senior Council Members SPLTAPPLSNOW, SPLTAPPLTARGARYEN, SPLTAPPLLANNISTER, OR SPLTAPPLSTARK OR even COB User COB_USER,  BVACOTBJUDGE, BVATCOBB (Must have "Hearing Prep" role).
4. Pull up the Appeal, through Queue, and Click the Correct issues button to begin the split appeal process. I Perform two intakes for Direct Review, Evidence Submission, and Hearing. Also under the two hearing Appeals (Conditional rendering logic was appealed if the hearing was scheduled or other dispostion types.)
![image](https://user-images.githubusercontent.com/96748497/191588134-6a3e4919-6e47-4c8a-9ba8-244b35101b67.png)
Image 1: Note how in creating test cases the appeal has a claimant for render logic downstream; in particular hearings; aditional logic will be presented below and I used the user Stacy Yellow to create the hearing view.
6. You will navigate to the last page "Review and Confirm Appeal Split Page"

### User Facing Changes
New Intake Edit Review and confirm new appeal stream split page ReviewAppealView.JSX.

![image](https://user-images.githubusercontent.com/96748497/191600911-d1563ecc-dd6c-40b2-8809-f0f9ad7d3ac2.png)
Note: Example of direct review without claimant

![image](https://user-images.githubusercontent.com/96748497/191601143-ff1cd3c7-7147-45b9-943a-60ff346b3b63.png)
Note: Example of direct review with claimant

![image](https://user-images.githubusercontent.com/96748497/191601398-c63abc1f-c07d-4004-8386-3179fcd68070.png)
Note: Example of evidence submission without claimant.

![image](https://user-images.githubusercontent.com/96748497/191601669-1ff79e64-7287-4d65-81b8-4e4166f25b3b.png)
Note: Example of evidence submission with claimant.

![image](https://user-images.githubusercontent.com/96748497/191602269-998b1e66-53f8-45e3-9651-9ebe2f814a4e.png)
Note: A unscheduled video hearing with no claimant example where no disposition.

![image](https://user-images.githubusercontent.com/96748497/191602572-c9264a1c-ae07-4b7d-b3e7-d9e6c2c8a9ee.png)
Note: A unscheduled virtual hearing with claimant example where no disposition.

*Assuming Central office hearings render. Now when hearing has been held or any other disposition by a Judge. The logic will display.

![image](https://user-images.githubusercontent.com/96748497/191603994-faf9553c-71df-4724-b28a-7df30a9d1070.png)
Example of a held virtual hearing with claimant.

![image](https://user-images.githubusercontent.com/96748497/191604566-3a76f2d1-b9a6-49bb-aa90-4243a8312c49.png)
Note: The hook information being sent to backend.

![image](https://user-images.githubusercontent.com/96748497/191604839-06b130d4-0f27-4f6c-be67-ed3125b8bfb7.png)
The hearing worksheet link renders on click if the Senior Supervisor Council or COB user with role "Hearing Prep"

### Code Documentation Updates

### Storybook Story
*For Frontend (Presentation) Components
* Capybara

### Database Changes: None
*Only for Schema Changes*: None

### Integrations: Adding endpoints for external APIs : No

* [  ] Integrations impacting functionality are tested in Caseflow UAT: Pending on Feature Branch APPEALS-3419
* db/seed on users.rb Added "Hearing Prep" Role
